server/src/main/java/org/elasticsearch/cluster/routing/allocation/command/MoveAllocationCommand.java

```diff
@@ -102,7 +102,16 @@ public RerouteExplanation execute(RoutingAllocation allocation, boolean explain)
         Decision decision = null;
 
         boolean found = false;
-        for (ShardRouting shardRouting : allocation.routingNodes().node(fromDiscoNode.getId())) {
+        RoutingNode fromRoutingNode = allocation.routingNodes().node(fromDiscoNode.getId());
+        if (fromRoutingNode == null && !fromDiscoNode.isDataNode()) {
+            throw new IllegalArgumentException("[move_allocation] can't move " + shardId + " from " + fromDiscoNode.getName() + " to " + toDiscoNode.getName() + ": " +  fromDiscoNode.getName() + " is not a data node.");
```

Comment:
Could you wrap these lines to <120 characters? We're trying to cut down on overlong lines.

Reply:
Sure thing!